### PR TITLE
Podman: add test kube apply

### PR DESCRIPTION
**Pull request by ilausuch**
- Related ticket:  https://progress.opensuse.org/issues/124601
VRs:
- SP4 http://openqa.suse.de/tests/10939764
- SP3 http://openqa.suse.de/tests/10947571
- SP2 http://openqa.suse.de/tests/10947574
- TW https://openqa.opensuse.org/tests/3233011

### CODE REVIEW

**Comment by ilausuch** [comment]
I changed the name for the pod from testing_pod to testing-pod, because:

`Error: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Pod \"testing_pod\" is invalid: [metadata.name: Invalid value: \"testing_pod\": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), ...`

**Comment by m-dati** [comment]
LGTM

**Review by ricardobranco777 — APPROVED** [review]
LGTM

**Review by grisu48 — COMMENTED** [review]
One suggestion, otherwise LGTM!

File: tests/containers/podman_pods.pm [hunk 1]
```diff
@@ -46,35 +47,41 @@ sub run {
...
         # Kube generate
         record_info('Test', 'Generate the yaml from a pod');
-        assert_script_run('podman pod create testing_pod');
+        assert_script_run('podman pod create testing-pod');
         my $image = "registry.suse.com/bci/bci-busybox:latest";
         script_retry("podman pull $image", timeout => 300, delay => 60, retry => 3);
-        assert_script_run("podman container create --pod testing_pod --name container $image sh -c \"sleep 3600\"");
-        assert_script_run("podman kube generate testing_pod | tee pod.yaml");
+        assert_script_run("podman container create --pod testing-pod --name container $image sh -c \"sleep 3600\"");
+        assert_script_run("podman kube generate testing-pod | tee pod.yaml");
```
[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/containers/podman_pods.pm:54]: Done, in a different way

[the remark this answers is not in the document]
Comment (ilausuch) [reply tests/containers/podman_pods.pm:54]: I done some similar for the kube apply. I understood this

**Review by jlausuch — APPROVED** [review]

File: tests/containers/podman_pods.pm [hunk 2]
```diff
@@ -46,35 +47,42 @@ sub run {
...
-        validate_script_output('podman ps', sub { !m/testing_pod-container/ });
+        validate_script_output('podman pod ps', sub { !m/testing-pod/ });
+        validate_script_output('podman ps', sub { !m/testing-pod-container/ });
+
+        # kube apply
+        install_k3s();
+        record_info('Test', 'kube apply');
+        assert_script_run('podman kube apply --kubeconfig ~/.kube/config -f pod.yaml');
+        assert_script_run('kubectl wait --for=condition=Ready pod/testing-pod');
+        validate_script_output('kubectl exec testing-pod -- cat /etc/os-release', sub { m/SUSE Linux Enterprise Server/ });
```
Comment (pdostal) [inline tests/containers/podman_pods.pm:85]: That looks super clean! 👍

**Review by pdostal — APPROVED** [review]

File: tests/containers/podman_pods.pm (hunk 2 above)
Comment (ilausuch) [reply tests/containers/podman_pods.pm:85]: Thanks

**Pull request merged**